pkg/provisioner/ironic/ironic.go

```diff
@@ -1319,3 +1357,102 @@ func (p *ironicProvisioner) PowerOff() (result provisioner.Result, err error) {
 
 	return result, nil
 }
+
+// A private method to set target for RAID configuration steps
+func (p *ironicProvisioner) setTargetRAIDCfg(ironicNode *nodes.Node) (err error) {
+	// Build target for RAID configuration steps
+	logicalDisks, err := p.buildTargetRAIDCfg()
+	if 0 == len(logicalDisks) || nil != err {
+		return
+	}
+	// Set target for RAID configuration steps
+	err = nodes.SetRAIDConfig(
+		p.client,
+		ironicNode.UUID,
+		nodes.RAIDConfigOpts{LogicalDisks: logicalDisks},
+	).ExtractErr()
+	if nil != err {
+		p.log.Info("failed in setting the target RAID configuration",
+			"logical disks", logicalDisks,
+			"error", err,
+		)
+	}
+	return
+}
+
+// A private method to build RAID disks
+func (p *ironicProvisioner) buildTargetRAIDCfg() (logicalDisks []nodes.LogicalDisk, err error) {
+	// If not configure raid
+	if nil == p.host.Spec.RAID || nil == p.host.Spec.RAID.Volumes || 0 == len(p.host.Spec.RAID.Volumes) {
+		return
+	}
+	// Build logicalDisks
+	var logicalDisk nodes.LogicalDisk
+	var rootVolumeFlag bool = false
+	for _, volume := range p.host.Spec.RAID.Volumes {
+		if true == volume.IsRootVolume {
+			if false == rootVolumeFlag {
+				rootVolumeFlag = true
+			} else {
+				err = errors.Errorf("most one logical disk can be set to root")
+				p.log.Info("failed in build the target RAID configuration",
+					"error", err,
+				)
+				return
+			}
+		}
+		logicalDisk = nodes.LogicalDisk{
+			RAIDLevel:             volume.Level,
+			IsRootVolume:          &volume.IsRootVolume,
+			VolumeName:            volume.Name,
+			DiskType:              volume.DiskType,
+			NumberOfPhysicalDisks: volume.NumberOfPhysicalDisks,
+		}
+		if 0 == volume.SizeGibibytes {
+			logicalDisk.SizeGB = nil
+		} else {
+			logicalDisk.SizeGB = &volume.SizeGibibytes
+		}
+		logicalDisks = append(logicalDisks, logicalDisk)
+	}
+	if false == rootVolumeFlag {
+		err = errors.Errorf("root volumes must be set")
+		p.log.Info("failed in build the target RAID configuration",
```

Comment:
Errors should be logged as errors, and this function should return only the error and an empty slice at this point.